### PR TITLE
feat: add support for fieldmask to documentreference

### PR DESCRIPTION
**Pull request by suraj-qlogic**

### CODE REVIEW

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java [hunk 1]
```diff
@@ -354,6 +354,19 @@ public T apply(List<T> results) {
     return extractFirst(firestore.getAll(this));
   }
 
+  /**
+   * Reads the document referenced by this DocumentReference. If the document doesn't exist, the
+   * get(FieldMask fieldMask) will return an an empty DocumentSnapshot.
```
Comment (dmitry-fa) [inline google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java]: an an -> an

**Review by pmakani — APPROVED** [review]

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java [hunk 2]
```diff
@@ -354,6 +354,19 @@ public T apply(List<T> results) {
...
+   *
+   * @param fieldMask A FieldMask Object for retrieve field value.
+   * @return An ApiFuture that will be resolved with the contents of the Document at this
+   *     DocumentReference, or a failure if the document does not exist.
+   */
```
Comment (dmitry-fa) [inline google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java:364]: the contents -> the content
**@param** <txt>. -> @param <txt>     (no . is required)
same for **@return**

Comment (dmitry-fa) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java:364]: I would say:
@param fieldMask a FieldMask object to retrieve the field value

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/DocumentReferenceTest.java [hunk 3]
```diff
@@ -275,6 +275,18 @@ public void deserializeDocumentReference() throws Exception {
...
+  public void getFieldWithFieldMask() throws Exception {
+    doAnswer(getAllResponse(SINGLE_FIELD_PROTO))
+        .when(firestoreMock)
+        .streamRequest(
+            getAllCapture.capture(),
+            streamObserverCapture.capture(),
+            Matchers.<ServerStreamingCallable>any());
+    DocumentSnapshot snapshot = documentReference.get(FieldMask.of(FieldPath.of("foo"))).get();
+    assertEquals("bar", snapshot.get("foo"));
+  }
```
Comment (dmitry-fa) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/DocumentReferenceTest.java:289]: You capture parameters:

```
            getAllCapture.capture(),
            streamObserverCapture.capture(),
```
Don't you need to check the captured values?

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java (hunk 2 above)
Comment (suraj-qlogic) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java:364]: Done

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/DocumentReferenceTest.java (hunk 3 above)
Comment (suraj-qlogic) [reply google-cloud-firestore/src/test/java/com/google/cloud/firestore/DocumentReferenceTest.java:289]: Done

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java (hunk 1 above)
Comment (suraj-qlogic) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java]: Done

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java [hunk 4]
```diff
@@ -354,6 +354,19 @@ public T apply(List<T> results) {
     return extractFirst(firestore.getAll(this));
   }
 
+  /**
+   * Reads the document referenced by this DocumentReference. If the document doesn't exist, the
+   * get(FieldMask fieldMask) will return an empty DocumentSnapshot.
+   *
```
Comment (dmitry-fa) [inline google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java:360]: As googlers say, the specification requires present tense 

If the document doesn't exist, the get(FieldMask fieldMask) will return an empty DocumentSnapshot.
->
If the document doesn't exist an empty DocumentSnapshot is returned.

**Review by dmitry-fa — APPROVED** [review]

**Pull request closed without merging**